### PR TITLE
Enable pending block tag in simulation endpoints

### LINKED ISSUE #1304: Enable pending block tag in simulation endpoints

### PR DESCRIPTION
**Pull request by exeokan**
# Description

## Linked Issues
- Fixes #1304

### CODE REVIEW

**Review by kpp — APPROVED** [review]

**Review by eyusufatik — APPROVED** [review]

**Comment by eyusufatik** [comment]
realized that pending = latest work for all things state but if a contract depends on block.number this would create problems. need to create a hypothetical `BlockEnv` and use that. it's a simple change

**Review by eyusufatik — CHANGES_REQUESTED** [review]

File: bin/citrea/tests/test_client/mod.rs [hunk 1]
```diff
@@ -656,7 +656,8 @@ impl TestClient {
         let end_block = match end_block {
             BlockNumberOrTag::Number(b) => b,
             BlockNumberOrTag::Latest => self.eth_block_number().await,
-            _ => panic!("Only number and latest"),
+            BlockNumberOrTag::Pending => self.eth_block_number().await + 1,
```
Comment (eyusufatik) [inline bin/citrea/tests/test_client/mod.rs:659]: forgot this?

File: crates/evm/src/query.rs [hunk 2]
```diff
@@ -1673,3 +1695,28 @@ fn set_state_to_end_of_evm_block<C: sov_modules_api::Context>(
     // so every block is offset by 1
     working_set.set_archival_version(block_number + 1);
 }
+
+fn get_pending_block_env<C: sov_modules_api::Context>(
```
Comment (eyusufatik) [inline crates/evm/src/query.rs:1699]: also need to set evm.latest_block_hashes here :)

**Review by eyusufatik — CHANGES_REQUESTED** [review]

File: crates/evm/src/tests/queries/estimate_gas_tests.rs [hunk 3]
```diff
@@ -363,6 +363,63 @@ fn estimate_gas_with_varied_inputs_test() {
     );
 }
 
+#[test]
+fn pending_env_test() {
```
Comment (rakanalh) [inline crates/evm/src/tests/queries/estimate_gas_tests.rs:367]: Convention `test_pending_env`

**Review by rakanalh — APPROVED** [review]
Small nit on test name

File: crates/evm/src/tests/queries/estimate_gas_tests.rs (hunk 3 above)
Comment (eyusufatik) [reply crates/evm/src/tests/queries/estimate_gas_tests.rs:367]: I'll change that once CI passes and merge

**Review by eyusufatik — APPROVED** [review]

**Pull request merged**